components/more_direct_channels/more_direct_channels.jsx

```diff
@@ -188,17 +190,31 @@ export default class MoreDirectChannels extends React.Component {
         } else {
             openGroupChannelToUsers(userIds, success, error);
         }
-    }
+    };
 
     addValue = (value) => {
-        const values = Object.assign([], this.state.values);
+        if (Array.isArray(value)) {
+            this.addUsers(value);
+        } else {
+            const values = Object.assign([], this.state.values);
 
-        if (values.indexOf(value) === -1) {
-            values.push(value);
+            if (values.indexOf(value) === -1) {
+                values.push(value);
+            }
+
+            this.setState({values});
         }
+    };
+
+    addUsers = (users) => {
+        const values = Object.assign([], this.state.values);
+        const existingUserIds = values.map((user) => user.id);
+        users.
+            filter((user) => existingUserIds.indexOf(user.id) === -1).
+            map((user) => values.push(user));
```

Comment:
This should use `forEach` instead of `map` since we're not using the returned value. It might also be more clear to use a regular for loop to skip the filter like
```javascript
for (const user of users) {
    if (existingUserIds.indexOf(user.id) !== -1) {
        // Already in values
        continue;
    }

    values.push(user);
}
```